### ISSUE TITLE
Support for vmware provider?

### ISSUE THREAD

**Issue by a-fro** [geerlingguy/packer-ubuntu-1404#2]
Hi Jeff, 

We're switching to vmware as a provider for performance reasons. I tried to install your drupal-dev-vm, which currently points to https://atlas.hashicorp.com/geerlingguy/boxes/ubuntu1404, but that only offers the virtualbox provider. I'm not sure what it would take to support another provider, or if this repo is the one that you deploy to the Atlas, but if it's not difficult to support vmware, I'd love you to add it.

Thanks!

**Comment by geerlingguy** [comment]
At this time, I don't have a license for VMWare, and haven't run into any issues with VirtualBox that would cause me to need to switch to Parallels or VMWare. However, you could find another Vagrant box that works with VMWare on Atlas and switch to that, hopefully with no issue (either 14.04 or 12.04).

If VMWare or Parallels would like to give me a free license or something, I might be tempted to take the time to get these packer profiles working with them, but as it is, it's enough work supporting one provider; I may also support AMIs in the future, or maybe Digital Ocean droplets, but no timelines for that.

As an aside; what performance issues were you seeing with VirtualBox? Besides graphics-related issues, I've rarely noticed more than a 5% difference in disk IO or CPU-related tasks... but last time I used Fusion was about 3 years ago, and last time I used Parallels was 2 years ago.

**Comment by a-fro** [comment]
The main issue was generally slow performance during development,
particularly on older OSX machines. A friend and long-time Drupal Dev,
Ethan Winn, recently switched to VMWare and found that it made a huge
difference for him, with 2x-3x performance improvement in his development
workflow. I don't have any benchmarks to prove that, but I trust him. You
can see another take here:
http://mitchellh.com/comparing-filesystem-performance-in-virtual-machines

Before ponying up for VMWare and the Vagrant/VMWare plugin, I tried
switching from NFS to rsync, but the transition was not seamless and
auto_rsync wasn't working correctly, so I abandoned that effort.

In case you're wondering, puphpet/ubuntu1404-x64 worked with VMWare and our
NFS setup, but chef/ubuntu-14.04 did not. I've only spent a few minutes on
the new setup, so don't have much to offer in terms of benchmarking or even
impressions (other than it _feels_ like it's running faster), but let me
know if you want me to report back after I've spent more time with it.

Thanks, as always, for all the great work you're doing for the community!

**Comment by geerlingguy** [comment]
Interesting; from my more outdated testing with NFS vs. native filesystems a couple years ago, I didn't see too significant a difference between VMWare and VirtualBox (in fact, both were often faster than native Mac OS X filesystem access... likely due to caching).

If you're using native shares, VMWare does have a 2x-3x edge over VirtualBox, but either way is slow as molasses compared to even NFS (which, again, is [2x-3x _slower_ than using rsync + native filesystems](http://www.midwesternmac.com/blogs/jeff-geerling/nfs-rsync-and-shared-folder) for most operations).

Regardless, I would be happy if someone that has VMWare would support a VMWare version of this box, but I can't allocate $150 to shell out to get a copy of Fusion Pro :(

**Comment by a-fro** [comment]
I took a brief look at the VMWare provider documentation at
https://docs.vagrantup.com/v2/vmware/index.html, which states that it is a
"drop-in replacement for VirtualBox". There are a few details on the
configuration
page https://docs.vagrantup.com/v2/vmware/configuration.html. Also, it
sounds like the issues I faced when trying the chef version were related to
what was described on the kernal upgrade page
https://docs.vagrantup.com/v2/vmware/kernel-upgrade.html.

Bottom line: I'm happy to create a PR to add support for the additional
provider, though it may take me a little time as I'm preparing for a trip
out of the country. If it sounds easy enough for you, given your
experience, I'm also happy to test it out on my box if you can add provider
support.

**Comment by geerlingguy** [comment]
I actually just bought VMWare Fusion, and will be trying to update all my base boxes to also build for VMWare... so more to come soon, hopefully!

**Comment by a-fro** [comment]
Great! Looking forward to checking it out.
